yt/frontends/enzo_p/io.py

```diff
@@ -48,19 +46,37 @@ def _read_field_names(self, grid):
                 message="Grid %s is missing from data file %s." %
                 (grid.block_name, grid.filename), ds=self.ds)
         fields = []
-        dtypes = set([])
+        ptypes = set()
+        dtypes = set()
+        # keep one field for each particle type so we can count later
+        sample_pfields = {}
         for name, v in iteritems(group):
             if not hasattr(v, "shape") or v.dtype == "O":
                 continue
             # mesh fields are "field <name>"
             if name.startswith("field"):
-                dummy, fname = name.split(" ", 1)
+                if self._sep is None:
+                    if " " in name:
+                        self._sep = " "
+                    else:
+                        self._sep = "_"
+                _, fname = name.split(self._sep, 1)
                 fields.append(("enzop", fname))
                 dtypes.add(v.dtype)
             # particle fields are "particle <type> <name>"
             else:
-                dummy, ftype, fname = name.split(" ", 2)
+                if self._sep is None:
+                    if " " in name:
+                        self._sep = " "
+                    else:
+                        self._sep = "_"
+                _, ftype, fname = name.split(self._sep, 2)
```

Comment:
Can you refactor to avoid the duplicated code here and on lines 58-63?

Reply:
Ah, I forgot I don't actually need this anymore since I'm no longer trying to maintain backward compatibility with this format change from a while ago. Will change, thanks!